koku/api/iam/view/user_preference.py

```diff
@@ -43,21 +44,23 @@ class UserPreferenceViewSet(mixins.CreateModelMixin,
     permission_classes = (IsObjectOwner, AllowAny)
     filter_backends = (DjangoFilterBackend,)
     filter_fields = ('name',)
+    schema = KokuPaginatedSchema()
 
     def get_queryset(self):
         """Get a queryset that only displays the owner's preferences."""
         queryset = models.UserPreference.objects.none()
-        req_user = self.request.user
+        req_user = None
+        if hasattr(self.request, 'user'):
```

Comment:
For here and below, did you find situations where `user` is not set? Or is this just something required for generation to behave appropriately?

Reply:
Also, this could be simplified to:

```
req_user = getattr(self.request, 'user', None)
```

Reply:
During schema generation, user does not get set.